brave/brave-ios
# Fix #5682: Fullscreen network picker

## Summary of Changes
- Implements the new full screen network picker UI
- Moved all logic for creating a new account when switching to a network without accounts (ex. Solana) into a store / view model and unit test it.
- Added test to `NetworkStoreTests` to verify chain list

This pull request fixes #5682

## Submitter Checklist:


## Test Plan:
1. Tap network picker button
2. Select a new network; verify modal closes upon selection
3. Open Wallet Settings -> Networks -> Enable show test networks
4. Tap network picker button
5. Tap detail disclosure button on Ethereum / Solana
6. Select a test network; verify modal closes upon selection
7. Tap network picker button
8. Tap 'Ethereum' / 'Solana' row; ; verify selection is mainnet and modal closes upon selection


## Screenshots:

![#5682 - 1](https://user-images.githubusercontent.com/5314553/179519816-8f138d5e-a7c9-4ba9-9951-478bbccff143.png) | ![#5682 - 2](https://user-images.githubusercontent.com/5314553/179519832-7a69f87c-df98-4946-8fa0-5ca59f9eb0f6.png) | ![#5682 - 3](https://user-images.githubusercontent.com/5314553/179519840-b7a32575-524d-42e0-bbb8-38de2080bdbc.png)
--|--|--



https://user-images.githubusercontent.com/5314553/179519880-78104f20-41ae-4639-bca8-b799bb882be7.mp4


## Reviewer Checklist:

  - `QA/(Yes|No)`
  - `bug` / `enhancement`
